src/ui/draw-workspace.c

```diff
@@ -139,6 +140,7 @@ draw_window (GtkWidget                   *widget,
 
               /* Give up. */
               if (icon_w > (winrect->width - 2) || icon_h > (winrect->height - 2))
+                cairo_surface_destroy (icon);
                 icon = NULL;
```

Comment:
You're missing parentheses, so you're effectively always setting `icon = NULL`, which means all workspace icons are gone. Similar to:
```
if (icon_w > (winrect->width - 2) || icon_h > (winrect->height - 2))
  {
    cairo_surface_destroy (icon);
  }
icon = NULL;
```
You need to change it to
```suggestion
              if (icon_w > (winrect->width - 2) || icon_h > (winrect->height - 2))
                {
                  cairo_surface_destroy (icon);
                  icon = NULL;
                }
```
(although I suspect the `icon = NULL` is redundant here, but shouldn't hurt)

Reply:
Doh! I hate braceless ifs :) Thanks for catching that, I'll fix!

FWIW, I don't think the `icon = NULL` part is redundant, as there is an `if (icon)` condition below.